py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,441 @@
+"""
+desitarget.lyazcat
+==============
+
+Post-redrock ML processing for LyA Quasar object identification.
+"""
+
+import os
+import numpy as np
+import sys
+from astropy.io import fits
+import fitsio
+import time
+
+from desitarget.geomask import match, match_to
+from desitarget.internal import sharedmem
+from desitarget import io
+from desispec.io import read_spectra
+
+from quasarnp.io import load_model
+from quasarnp.io import load_desi_coadd
+from quasarnp.utils import process_preds
+from squeze.model import Model
+from squeze.common_functions import save_json, load_json
+from squeze.candidates import Candidates
+from squeze.desi_spectrum import DesiSpectrum
+from squeze.spectra import Spectra
+
+# ADM set up the DESI default logger.
+from desiutil.log import get_logger
+log = get_logger()
+
+zcatdatamodel_names = np.array(['TARGETID','Z','ZWARN','DELTACHI2','Z_COMB',
+                                'Z_COMB_PROB','Z_QN','Z_QN_CONF','IS_QSO_QN',
+                                'Z_SQ','Z_SQ_CONF','Z_ABS','Z_ABS_CONF'])
+"""
+zcatdatamodel_formats = np.array(['>int64','>float64','>int64','>float64',
+                                  '>float64','>float64','>float64','>float64',
+                                  '>int16','>float64','>float64','>float64',
+                                  '>float64'])
+"""
+zcatdatamodel_formats = np.array(['>i8','>f8','>i8','>f8',
+                                  '>f8','>f8','>f8','>f8',
+                                  '>i2','>f8','>f8','>f8',
+                                  '>f8'])
+zcols_copy = np.array(['TARGETID','Z','ZWARN','DELTACHI2'])
+n1_cols = np.array(['Z_QN','Z_QN_CONF','IS_QSO_QN',
+                    'Z_SQ','Z_SQ_CONF','Z_ABS','Z_ABS_CONF'])
+
+
+def tmark(istring):
+    """A function to mark the time an operation starts or ends.
+    
+    Parameters
+    ----------
+    istring : :class:'str'
+        The input string to print to the terminal.
+        
+    Output
+    ------
+    :class:'str'
+        A string with the date and time in ISO 8061 standard followed
+        by the istring.
+    """
+    t0 = time.time()
+    t_start = time.strftime('%Y-%m-%d | %H:%M:%S')
+    print('\n{}: {}'.format(istring,t_start))
```

Comment:
Replace any `print` statements with `log.info` to use the DESI logger.

Reply:
I'm using my version, for now, because it prints out date and time codes so I can see how long each step takes. Once the script is nailed down I'll remove this function (and all calls for it) and convert to the DESI logger.

Reply:
If you just replace `print` with `log.info`, your bespoke function should still work. So, there's no need to remove the function and the calls to it. Just switch over to logging instead of printing.

Reply:
I'll check this tonight when I make the MgII absorption additions and the logger changeover.

Reply:
Switched to using the logger.